scan_engine_raw.cc

```diff
@@ -747,9 +747,13 @@ int get_ping_pcap_result(UltraScanInfo *USI, struct timeval *stime) {
     } else if (hdr.proto == IPPROTO_TCP && USI->ptech.rawtcpscan) {
       struct tcp_hdr *tcp = (struct tcp_hdr *) data;
       /* Check that the packet has useful flags. */
-      if (!(tcp->th_flags & TH_RST)
-          && ((tcp->th_flags & (TH_SYN | TH_ACK)) != (TH_SYN | TH_ACK)))
-        continue;
+      if (o.host_discovery_ignore_tcp_reset
```

Comment:
Should this also affect TCP Connect host discovery in `scan_engine_connect.cc`? We would avoid setting `newhoststate = HOST_UP` under `case ECONNREFUSED:` if `USI->ping_scan && o.discovery_ignore_rst`.

If we don't make this additional change, we should be very clear in the docs that the option only affects raw TCP host discovery probes, not unprivileged TCP host discovery. (There's not a way to use TCP Connect for host discovery if you're privileged; it just uses raw SYN for -PS, and -PT means ACK.)